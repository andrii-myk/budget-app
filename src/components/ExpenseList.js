import React from 'react'
import { connect } from 'react-redux'

import ExpenseListItem from './ExpenseListItem'
import selectExpenses from './../selectors/expenses'

export const ExpenseList = props => (
    <div>{
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                renderList(props.expenses)
            )
         }  
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

const renderList = expenses => (
    expenses.map(expense => {
        return (
            <ExpenseListItem key={expense.id} {...expense}/>
        )
    })
)
export default connect(mapStateToProps)(ExpenseList)
