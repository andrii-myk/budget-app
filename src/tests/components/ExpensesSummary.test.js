import React from 'react'
import { shallow } from 'enzyme'

import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpenseSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9495} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={124435} />)
    expect(wrapper).toMatchSnapshot()
})