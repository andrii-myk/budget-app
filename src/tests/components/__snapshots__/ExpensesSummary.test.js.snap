// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h2>
    Sum of 
    1
     
    expense
     is 
    $94.95
  </h2>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Sum of 
    3
     
    expenses
     is 
    $1,244.35
  </h2>
</div>
`;
